lms/djangoapps/course_api/blocks/api.py

```diff
@@ -52,6 +53,7 @@ def get_blocks(
     # create ordered list of transformers, adding BlocksAPITransformer at end.
     transformers = BlockStructureTransformers()
     include_special_exams = False
+    include_completion = requested_fields is not None and 'completion' in requested_fields
     if requested_fields is not None and 'special_exam_info' in requested_fields:
         include_special_exams = True
```

Comment:
Nit: There are multiple lines here checking if requested_fields is not None.  Also, include_completion is injected into the middle of the logic for checking special_exams (L54 & LL57-58).  Consider changing LL54-58 to

    if requested_fields is None:
        requested_fields = []
    include_completion = 'completion' in requested_fields
    include_special_exams = 'special_exam_info' in requested_fields

Reply:
👍